### ISSUE TITLE
Local search not working with i18n

### ISSUE THREAD

**Issue by flohaider** [vuejs/vitepress#2371]
### Describe the bug

For my simple documentation site with 2 languages (`en` and `de`) and search functionality, everything works perfectly except when I try to search in any locale but my root. My root locale is english and my 2nd locale is german, with it's markdown files in the `de` folder.

The search results on all german sites are always empty, no error in console.

### Reproduction

Create a simple vitepress project with `srcDir` configured as `./src` and 2 markdown files in the `src` folder:
- Content of `./src/index.md`
```markdown
# English
This is the english start page
```
- Content `./src/de/index.md`
```markdown
# Deutsch
Das ist die deutsche Startseite
```

Finally configure vitepress like so:
```ts
import { defineConfig } from 'vitepress'

export default defineConfig({
	srcDir: './src',
	themeConfig: {
		search: {
			provider: 'local'
		}
	},
	locales: {
		root: {
			label: 'English',
			lang: 'en'
		},
		de: {
			label: 'Deutsch',
			lang: 'de'
		}
	}
})
```

Now searching on the root path works as it should:
<img width="898" alt="english" src="https://github.com/vuejs/vitepress/assets/6113939/249b19f5-3e19-43e4-8efe-7825c14c6448">

But after navigating to the `/de` path, search stops delivering results:
<img width="899" alt="german" src="https://github.com/vuejs/vitepress/assets/6113939/1a90a94e-ebc3-420b-8fad-7cbf3cf6f7d0">

### Expected behavior

Local search should work the same no matter what language my site is on.

Ideally searching in english should only deliver english results and searching in german should only bring german results :-)
But for now I would be happy if search works at all on my german pages.

### System Info

```sh
System:
    OS: Windows 10 10.0.22621
    CPU: (6) x64 Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
    Memory: 19.84 GB / 31.93 GB
Binaries:
    Node: 16.15.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.5.5 - C:\Program Files\nodejs\npm.CMD
Browsers:
    Chrome: 113.0.5672.93
    Edge: Spartan (44.22621.1702.0), Chromium (113.0.1774.35)
    Internet Explorer: 11.0.22621.1
npmPackages:
    vitepress: ^1.0.0-alpha.75 => 1.0.0-alpha.75
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Ah weird, we tested this before and it was working then (_searching in english should only deliver english results and searching in german should only bring german results_). Might be a regression. Can you check if it works when srcDir is not specified?

**Comment by zonemeen** [comment]
This bug was fixed in 1.0.0-alpha.76, see this [pr](https://github.com/vuejs/vitepress/pull/2358), it was still caused by windows slash.😂

@flohaider You can update the latest version of vitepress to solve this problem.
